os/board/common/partitions.c

```diff
@@ -110,77 +192,16 @@ void configure_partitions(void)
 			lldbg("ERROR: failed to create partition.\n");
 			return;
 		}
-#ifdef CONFIG_MTD_FTL
-		if (!strncmp(types, "ftl,", 4)
-#ifdef CONFIG_FS_ROMFS
-		|| !strncmp(types, "romfs,", 6)
-#endif
-#ifdef CONFIG_BINARY_MANAGER
-		|| !strncmp(types, "kernel,", 7) || !strncmp(types, "bin,", 4)
-#endif
-		) {
-			if (ftl_initialize(partno, mtd_part)) {
-				lldbg("ERROR: failed to initialise mtd ftl errno :%d\n", errno);
-				return;
-			}
-		} else
-#endif
 
-#ifdef CONFIG_MTD_CONFIG
-		if (!strncmp(types, "config,", 7)) {
-			mtdconfig_register(mtd_part);
-		} else
-#endif
-#if defined(CONFIG_MTD_SMART) && defined(CONFIG_FS_SMARTFS)
-		if (!strncmp(types, "smartfs,", 8)) {
-			char partref[4];
+		type_specific_initialize(mtd_part, partno, types);
 
-			snprintf(partref, sizeof(partref), "p%d", partno);
-			smart_initialize(CONFIG_FLASH_MINOR, mtd_part, partref);
-		} else
-#endif
-		{
-		}
-
-#if defined(CONFIG_MTD_PARTITION_NAMES)
-		if (*names) {
-			while (*names) {
-				if (*names == ',') {
-					part_name[index] = '\0';
-					index = 0;
-					names++;
-					break;
-				}
-				if (index < MTD_PARTNAME_LEN) {
-					part_name[index++] = *(names++);
-				} else {
-					part_name[index] = '\0';
-					index = 0;
-					lldbg("ERROR: Partition name is so long. Please make it smaller than %d\n", MTD_PARTNAME_LEN);
-					/* Move to next part name information. */
-					while (*(names++) != ',');
-					break;
-				}
-			}
-			mtd_setpartitionname(mtd_part, part_name);
+#ifdef CONFIG_MTD_PARTITION_NAMES
+		configure_partition_name(mtd_part, &names, &index, part_name);
 #ifdef CONFIG_BINARY_MANAGER
-			if (!strncmp(types, "kernel,", 7)) {
-				binary_manager_register_partition(partno, BINMGR_PART_KERNEL, part_name, partsize);
-			} else if (!strncmp(types, "bin,", 4)) {
-				binary_manager_register_partition(partno, BINMGR_PART_USRBIN, part_name, partsize);
-			}
+		binary_manager_register(partno, part_name, partsize, types);
 #endif
-		}
-
 #endif
-
-		/* Move to next part size information. */
-		while ((*parts++) != ',');
-
-		/* Move to next part type information. */
-		while ((*types++) != ',');
-
-		partno++;
+		move_to_next_part(&parts, &types, &partno);
```

Comment:
`
static void move_to_next_part(const char **parts)
{
		/* Move to next  information. */
		while (*(*parts)++ != ',');
}
`
I think this is better.
After it is good to use in while (*(names++) != ',');, while ((*parts++) != ','); and while ((*types++) != ',');.
Like below
`
move_to_next_part(&name);
move_to_next_part(&parts);
move_to_next_part(&types);
`

Reply:
I agree with your opinion.
I will fix it.